File: Source/Api/AuthenticatedUser/AuthenticatedUser.cs
using System;
using System.Collections.Generic;
using Affecto.AuthenticationServer.Claims;

namespace ServiceRegister.Api.AuthenticatedUser
{
    public class AuthenticatedUser : Affecto.AuthenticationServer.Claims.AuthenticatedUser
    {
        public Guid OrganizationId { get; private set; }

        public AuthenticatedUser(Guid id, string name, string accountName, Guid organizationId, IEnumerable<string> permissions = null, IEnumerable<string> roles = null,
            IEnumerable<CustomProperty> customProperties = null)
            : base(id, name, accountName, permissions, roles, customProperties)
        {
            if (organizationId == Guid.Empty)
            {
                throw new ArgumentException("Organization id must be provided.", "organizationId");
            }

            OrganizationId = organizationId;
        }
    }
}

File: Source/AngularApplication/App/Models/AuthenticatedUser.ts
"use strict";

module ServiceRegister
{
    export class AuthenticatedUser extends Affecto.Login.AuthenticatedUser
    {
        constructor(public name: string, public accountName: string, public roles: Array<string>, public permissions: Array<string>,
            public customProperties: Array<Affecto.Login.CustomProperty>, public organizationId: string)
        {
            super(name, accountName, roles, permissions, customProperties);
        }
    }
}